compiler/haskell-ide-core/src/Development/IDE/State/FileStore.hs

```diff
@@ -6,44 +6,74 @@
 module Development.IDE.State.FileStore(
     getFileExists, getFileContents,
     setBufferModified,
-    fileStoreRules
+    fileStoreRules,
+    VFSHandle(..),
+    makeVFSHandle,
+    makeLSPVFSHandle,
     ) where
 
 
 
 import           StringBuffer
 import Development.IDE.Orphans()
 
+import Control.Concurrent.STM
 import qualified Data.Map.Strict as Map
+import Data.Maybe
 import qualified Data.Text as T
 import           Data.Time.Clock
 import           Control.Monad.Extra
 import qualified System.Directory as Dir
 import           Development.Shake
 import           Development.Shake.Classes
 import           Development.IDE.State.Shake
-import           Control.Concurrent.Extra
 import           Control.Exception
 import           GHC.Generics
 import System.IO.Error
 import qualified Data.ByteString.Char8 as BS
 import qualified StringBuffer as SB
 import Development.IDE.Types.Diagnostics
+import qualified Data.Rope.UTF16 as Rope
 import           Data.Time
 
-
--- This module stores the changed files in memory, and answers file system questions
--- from either the memory changes OR the file system itself
-
-type DirtyFiles = Map.Map FilePath (UTCTime, StringBuffer) -- when it was modified, it's current value
-
--- Store the DirtyFiles globally, so we can get at it through setBufferModified
-newtype GlobalDirtyFiles = GlobalDirtyFiles (Var DirtyFiles)
-instance IsIdeGlobal GlobalDirtyFiles
+import Language.Haskell.LSP.Core
+import Language.Haskell.LSP.VFS
+
+-- | haskell-lsp manages the VFS internally and automatically so we cannot use
+-- the builtin VFS without spawning up an LSP server. To be able to test things
+-- like `setBufferModified` we abstract over the VFS implementation.
```

Comment:
I'm not averse to spinning up an LSP server to test, then our tests are more like real. We'd then just mock VS Code. Although happy to go this way for now.

Reply:
Yeah, I am certainly planning to do LSP based tests (probably using `lsp-test`). That would also allow us to finally kill those dreaded language server tests. I just didn’t want to also cram this in this PR as well and keeping `daml-ghc-shake-test-ci` intact for now seems useful.